### PR TITLE
Theme watch

### PR DESCRIPTION
**Pull request by edwtorba**
## Description:
- Implements watch command.

## Steps to test:

- [ ] Download a project.
- [ ] Configure ddev:
  - [ ] Run: `ddev config`
  - [ ] Run: `ddev start`
  - [ ] Run: `ddev composer install`
  - [ ] Alter the fire.yml file to use ddev as local env.
  - [ ] Run: `fire setup -y`
- [ ] Install this fire branch into the project:
  - [ ] Run: `ddev composer require fourkitchens/fire:dev-feature/v2-theme-improvements --dev`
  - [ ] Add the parameter: "local_theme_watch_script" to the fire.yml file.
- [ ] Test the theme watch command:
  - [ ] Run: `fire theme-watch`
  - [ ] Check the command wait for Styles changes.

### CODE REVIEW

File: src/Robo/Plugin/Commands/ThemeWatchCommand.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    switch ($env) {
+      case 'lando':
+      default:
+        if (file_exists($root . '/.nvmrc') && getenv('NVM_DIR')) {
+          $command = 'export NVM_DIR=$HOME/.nvm && . $NVM_DIR/nvm.sh && cd ' . $root . ' && nvm install && npm ci && npm run ' . $npmCommand;
+        }
+        break;
+      case 'ddev':
+        if (file_exists($root . '/.nvmrc') && getenv('NVM_DIR')) {
+          $command = 'cd ' . $root . ' && ddev nvm install && ddev npm ci && ddev npm run ' . $npmCommand;
```
Comment (rigoucr) [inline src/Robo/Plugin/Commands/ThemeWatchCommand.php]: For me this setup with ddev is not working at least for the emulsify npm install:
Screenshot 2024-08-31 at 2 36 04 PM

We might want to run this just like for lando

File: src/Robo/Plugin/Commands/ThemeWatchCommand.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+   *
+   * @command local:theme:watch
+   * @aliases theme-watch, watch-theme, tw
+   *
+   */
+  public function themeWatch(ConsoleIO $io) {
+    $env = Robo::config()->get('local_environment');
+    $root = $this->getThemePath();
+    $tasks = $this->collectionBuilder($io);
+    $npmCommand = Robo::config()->get('local_theme_watch_script') ?: 'watch';
```
Comment (rigoucr) [inline src/Robo/Plugin/Commands/ThemeWatchCommand.php]: 1 - Please add this to the  fire.yml template
2- add a option for this into the new Fire init Wizard
3 - Setup the default value into the wizard and template, not over here

**Review by rigoucr — CHANGES_REQUESTED** [review]
Thanks Ed, almost there , but there are some details that needs to be fixed .

Also please add to the Readme the new command and the Configuration variable .

File: src/Robo/Plugin/Commands/ThemeWatchCommand.php (hunk 1 above)
Comment (edwtorba) [reply src/Robo/Plugin/Commands/ThemeWatchCommand.php]: Test it and works fine.

File: src/Robo/Plugin/Commands/ThemeWatchCommand.php (hunk 2 above)
Comment (edwtorba) [reply src/Robo/Plugin/Commands/ThemeWatchCommand.php]: Everything is done!

**Review by rigoucr — CHANGES_REQUESTED** [review]
@edwtorba a minor code change needed

**Review by rigoucr — APPROVED** [review]
Thx @edwtorba , approved

**Pull request merged**